services/visual_recognition/v3.js

```diff
@@ -62,301 +37,329 @@ module.exports = function (RED) {
   RED.httpAdmin.get('/watson-visual-recognition/vcap', function (req, res) {
     res.json(service ? {bound_service: true} : null);
   });
-
-  // Utility functions that check for image buffers, urls and stream data in
  
   function imageCheck(data) {
     return data instanceof Buffer && imageType(data) !== null;
-  };
+  }
 
   function urlCheck(str) {
     var parsed = url.parse(str);
     return (!!parsed.hostname && !!parsed.protocol && str.indexOf(' ') < 0);
-  };
+  }
 
   function stream_buffer(file, contents, cb) {
     fs.writeFile(file, contents, function (err) {
-      if (err) throw err;
+      if (err) {
+        throw err;
+      }
       cb();
     });
-  };
+  }
+
+  function verifyPayload(node, msg) {
+    if (!msg.payload) {
+      this.status({fill:'red', shape:'ring', text:'missing payload'}); 
+      node.error('Missing property: msg.payload', msg);
+      return false;
+    }
+    return true;
+  }
+
+  function verifyInputs(feature, node, msg) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Function 'verifyInputs' has a complexity of 6.](https://www.codacy.com/app/node-red-nodes-watson/node-red-node-watson/file/2737057949/issues/source?bid=3225115&fileBranchId=3352493#l68)